clients/src/main/java/org/apache/kafka/common/network/KafkaChannel.java

```diff
@@ -363,9 +363,9 @@ public InetAddress socketAddress() {
 
     public String socketDescription() {
         Socket socket = transportLayer.socketChannel().socket();
-        if (socket.getInetAddress() == null)
-            return socket.getLocalAddress().toString();
-        return socket.getInetAddress().toString();
+        return "(Remote Address: " + (
+            socket.getInetAddress() == null ? "[non-connected socket]" : socket.getInetAddress().toString()
```

Comment:
We'll get NPE here if `socket` is null, right?

Reply:
Yes. I assume we don't get this since this code has lived for quite a while, so I wasn't defensive about it.
What do you think we should return in the scenario where this is null? `[non-existing socket]`?

Reply:
sounds good to me :)